# Relatório de progresso diário
Este relatório tem o objetivo de demonstrar a importância da análise dos dados de um projeto de desenvolvimento de software ao longo de uma semana. Os dados fornecidos permitirão ao proprietário da equipe de desenvolvimento obter insights sobre o progresso do projeto, identificar possíveis áreas de melhoria e tomar decisões informadas para garantir o sucesso do projeto.

## Bibliotecas

In [10]:
import pandas as pd
import numpy as np

## Dados
Os dados foram fornecidos em uma tabela com as seguintes colunas: dia, horas trabalhadas, bugs corrigidos e tarefas concluídas.

In [2]:
dados = {
    'Dia': ['Segunda', 'Terça', 'Quarta', 'Quinta', 'Sexta', 'Sábado', 'Domingo'],
    'Horas Trabalhadas': [6, 7, 8, 6, 7, 5, 4],
    'Bugs Corrigidos': [3, 2, 1, 4, 3, 2, 1],
    'Tarefas Concluídas': [5, 4, 6, 4, 5, 3, 2]
}

df = pd.DataFrame(dados)

## Análise Exploratória
 
 Textinho

In [5]:
total_horas_trabalhadas = df['Horas Trabalhadas'].sum()
print(f'Total de horas trabalhadas: {total_horas_trabalhadas}')

media_horas_trabalhadas = df['Horas Trabalhadas'].mean()
print(f'Média de horas trabalhadas: {round(media_horas_trabalhadas,2)}')

total_bugs_corrigidos = df['Bugs Corrigidos'].sum()
print(f'Total de bugs corrigidos: {total_bugs_corrigidos}')

media_tarefas_concluidas = df['Tarefas Concluídas'].mean()
print(f'Média de bugs corrigidos: {round(media_tarefas_concluidas,2)}')

Total de horas trabalhadas: 43
Média de horas trabalhadas: 6.14
Total de bugs corrigidos: 16
Média de bugs corrigidos: 4.14


## Produtividade diária (tarefas concluídas por hora)
Textinho

In [14]:
produtividade = df['Tarefas Concluídas'] / df['Horas Trabalhadas']
df['Produtividade Diária'] = round(produtividade, 2)
df

Dia  Horas Trabalhadas  Bugs Corrigidos  Tarefas Concluídas  \
0  Segunda                  6                3                   5   
1    Terça                  7                2                   4   
2   Quarta                  8                1                   6   
3   Quinta                  6                4                   4   
4    Sexta                  7                3                   5   
5   Sábado                  5                2                   3   
6  Domingo                  4                1                   2   

   Produtividade Diária  
0                  0.83  
1                  0.57  
2                  0.75  
3                  0.67  
4                  0.71  
5                  0.60  
6                  0.50

## Resultados e Insights

### Total de horas trabalhadas e bugs registrados
O total de horas trabalhadas na semana foi de **43 horas**.
O total de bugs corrigidos na semana foi de **16 bugs**.

### Total de tarefas concluídas e bugs corrigidos
A média diária de horas trabalhadas foi de **6.14 horas**.
A média diária de bugs corrigidos foi de **2.29 bugs**.

O total de tarefas concluídas na semana foi de **29 tarefas**.

### Média Diária de Tarefas Concluídas
A média diária de tarefas concluídas foi de 4.14 tarefas.

## Produtividade Diária
A produtividade diária, medida em tarefas concluídas por hora trabalhada, variou ao longo da semana, com picos nos dias em que a quantidade de tarefas concluídas foi maior em relação às horas trabalhadas.

## Conclusão
Este relatório fornece uma visão detalhada do progresso diário da equipe de desenvolvimento de software. A análise dos dados mostra que, embora a produtividade tenha variado ao longo da semana, houve um esforço consistente para corrigir bugs e concluir tarefas. Essas informações são essenciais para identificar padrões de trabalho, áreas de melhoria e tomar decisões informadas para otimizar o desempenho da equipe e o sucesso do projeto.

## Recomendações
Manter um registro constante e detalhado das horas trabalhadas, bugs corrigidos e tarefas concluídas.
Analisar regularmente os dados para identificar tendências e áreas de melhoria.
Considerar ajustar a carga de trabalho ou alocar recursos de forma mais eficaz com base nas observações feitas a partir dos dados analisados.